### DISCUSSION TITLE
Rock 5B

### DISCUSSION THREAD

**Discussion by tyrellm04** [Joshua-Riek/ubuntu-rockchip#631] [General]
Hello,

Please forgive me if this is addressed somewhere, and then after forgiving me, please point me in that direction.

I have a Rock 5B... the Rock Build 39 debian image is... unusable I've had to reflash 6x due to little gotchas putting it into a bootloop. That has me looking at this. I've already used the Rockdevtools to flash the SPI to boot from NVME and have successfully booted from NVMe 6x with their Rock build 39 image. 

I followed the directions in the read me and downloaded the current Rock 5B build from https://joshua-riek.github.io/ubuntu-rockchip-download/boards/rock-5b.html, I used Etcher to burn it to an SSD, it validated successfully, however, I can't even get any output from the HDMI to a monitor on either of the 2 NVMe drives that I flashed with this Ubuntu image. I know that the read me says the desktop version requires following the wizard in the UI, so obviously SSH won't work, but I'm not even seeing it as showing up in my router has having pulled an IP.

This has me wondering if I did something wrong, though I'm unsure how since it's very basic and only 2 steps 🤣

I know that the first question that will come up is the power delivery. I've tried both the official USB-C and the Radxa PoE Hat, both of which boot the NVME with the Build 39 on it.

![balenaEtcher_Oxb573bxXW](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/79353114/e2d9ec7e-263f-4c09-8f1f-eb74a0ed35ab)
![2024-02-22 09_01_57-RKDevTool v2 96](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/79353114/2b3b7e97-3eda-4588-8563-f6ee2487e6f3)

**Comment by Joshua-Riek** [comment]
Your boot loop issue is because of broken PD. You need to use a static 5v 4a power supply. 

Please boot from an SD card first and enter the below command to flash the SPI so you can boot from NVMe:
```
sudo u-boot-install-mtd /dev/mtdblock0
```
If you can connect a serial console to the board and post the output information I will be able to determine what the issue is.

**Reply by tyrellm04** [reply]
Thanks for the quick reply Joshua, however, in my initial post I state that I'm using the Radxa official USB C power supply, as well as the official Radxa PoE hat, both of which should be supplying the appropriate power, right? Also, I'm successfully booting, but the gotcha's I previously ran into were related to apt updates/upgrades periodically throwing it into a bootloop, even after getting their cert and approving their repo.

As for the SPI, isn't that what's depicted above in my RKDevtools screenshot? If I'm already able to successfully boot to their Rock build 39 Debian off of NVME, shouldn't I be able to boot any image from NVME since the SPI has been flashed via rkdevtools?

I don't have the console cable, and may end up getting one to test some of this, but here are a couple of pics of me booted via NVME to using their unstable image and showing that the rock has no SD card in, just the NVME

![RockBuild39](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/79353114/8ceb8d64-eef4-4d90-b467-9a73b18d988f)
![Rock 5B](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/79353114/fd0e3299-670d-47bb-bdba-0d2e2952cf63)

**Reply by Joshua-Riek** [reply]
The official power supply from radxa uses USB-C PD which is pretty much broken. 

> As for the SPI, isn't that what's depicted above in my RKDevtools screenshot? If I'm already able to successfully boot to their Rock build 39 Debian off of NVME, shouldn't I be able to boot any image from NVME since the SPI has been flashed via rkdevtools?

The SPI image you flashed is not compatible with this distro you need to update it. You need to install the OS onto an SD card and get to a command prompt.  

After that enter the command:
```
sudo u-boot-install-mtd /dev/mtdblock0
```

**Reply by tyrellm04** [reply]
Got it!!!! Thank you for the clarification and helping me learn. I will see about getting the spi flashed after my morning meetings.

**Reply by tyrellm04** [reply]
Well, I  got SOOOOOOOO Close to being able to do the u-boot-install-mtd /dev/mtdblock0 command... but immediately after finishing the setup guide boot off of an SD card, I saw it load the desktop briefly, as I moved over to the bench where it's being worked on It crashed the UI and presented this.

![RockUbuntuCrash](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/79353114/52070bf0-7e0e-4c17-867f-f81e33cc706b)

I thought maybe a reboot would fix it.... but I just get hung at a different screen now.
![RockUbuntuBootFailure](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/79353114/21b828e5-445e-4007-8e38-cd1a957592f0)

Do you think it would be best to use the Ubuntu server image, since SSH is enabled by default and there's no UI to crash, and then I can SSH to it to perform the u-boot command from my PC?

**Reply by Joshua-Riek** [reply]
The crash could be because of the SPI image you flashed, causing ram instability, or it's because of your power adapter. 

I'd try to erase the spi flash or the server image.

**Reply by tyrellm04** [reply]
Will do. When you say to erase the SPI, can I just use the rkdevtool and select the option "clear" on the far right to remove the SPI entirely? Or is there a different/better way to do that without the use of a console cable that I don't yet have?

**Reply by Joshua-Riek** [reply]
I use Linux and not windows so I don't know the exact steps on windows. But you can simply use dd to zero the first 4MB of the spi

**Reply by tyrellm04** [reply]
You sir are a hero and proof that no all heroes wear capes. I was able to successfully install the server OS on an SD, then SSH to it and run the SPI flash.  From there, I was able to remove the SD card, hook up the NVMe, and it booted right in, went past the wizard and is now humming away.


Thank you very much for your help, for providing this package and for helping me learn. I think Pi was a great set of training wheels for me, but this felt like someone took them off for the first time, so I appreciate your patience and direction.

**Reply by Joshua-Riek** [reply]
I do need to add some troubleshooting documentation, but I'm happy it's working now.

**Reply by tyrellm04** [reply]
I've been documenting some of my journey as I've gone, so if you'd like me to send you over a write up, I'm happy to do so. I do think that much of my trouble came from both having pre-flashed the SPI with the Radxa tool and image before trying your Ubuntu image, having limited SBC/Linux experience and not having all of the tools that would have made my life easier... like a console cable 🤣

When I saw you step about flashing the SPI, I ignorantly thought this was akin to setting up boot order in a PC BIOS... and said "Yup, did that" without realizing that the image set there needed to match the OS image being configured.

**Comment by horatio1989** [comment]
Hey! I think I am having the exact same issue!

I flashed the Loader and U-Boot (both from the Raxda website for the Rock 5B), and flashed the Ubuntu image to the eMMC. But every time the Rock boots, loads for a couple of seconds Ubuntu (still needs to be installed...), then go to the shell screen with all the same errors as you, and finally boots again. Thus, the cycle continues.

If I flash Ubuntu to an SD, and plug the SD, it is exactly the same (even if I unplug the eMMC).

I think my main issue Is that I am trying to install Ubuntu, after an experiment on which I installed W11 on the Rock 5B, and for that I needed UEFI instead of U-Boot.

Would you mind sharing your steps and sources of files? I have already tried everything that I could. I even installed a VIrtualBox with Ubuntu on my W11 just to be able to compile my own SPL and U-Boot, but that didn't solve the issue, and it is behaving the same.

EDIT:
nevermind, the trick was to use a dummy Power Supply...
